java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java

```diff
@@ -254,6 +288,11 @@ private void writeFromVectorIntoJson(Field field, FieldVector vector) throws IOE
         }
         generator.writeEndArray();
       }
+      if (variadicBufferCount == 0 && vector instanceof BaseVariableWidthViewVector) {
```

Comment:
Why so many special cases? Above if `!isFixedBufferCount` just have it always write out the variadic buffers. You already check if `dataBuffers.isEmpty()`.

Reply:
Not required. I removed.